services/QuillLMS/spec/controllers/teachers/progress_reports_controller_spec.rb

```diff
@@ -281,6 +281,33 @@
     end
   end
 
+  describe '#skills_report' do
+    let(:visible_classroom) { create(:classroom, visible: true) }
+    let!(:classrooms_teacher1) { create(:classrooms_teacher, classroom: visible_classroom, user: teacher) }
+    let(:level_two_concept) { create(:concept) }
+    let!(:level_one_concept) { create(:concept, parent: level_two_concept) }
+
+    it 'should render skills_report' do
+      get :skills_report
+      expect(response).to render_template 'skills_report'
+    end
+
+    it 'should assign classrooms to the classrooms that belong to the current_user' do
+      get :skills_report
+      expect(assigns(:classrooms)).to eq [visible_classroom]
+    end
+
+    it 'should assign all_skill_categories to level 2 concepts' do
+      get :skills_report
+      expect(assigns(:all_skill_categories)).to eq [level_two_concept]
+    end
+
+    it 'should assign all_skill_categories to level 1 concepts' do
+      get :skills_report
+      expect(assigns(:all_skill_sub_categories)).to eq [level_one_concept]
+    end
```

Comment:
I don't know that this is actually a better pattern, but all of these `assigns` tests could be wrapped in a `context` with the `get` call in a `before` block to reduce repitition.
```ruby
context 'assigns checks' do
  before { get :skills_report }

  it { expect(assigns(:classrooms)).to eq [visible_classrooms] }
  ... etc.
end
```